opensfm/pairs_selection.py

```diff
@@ -81,70 +31,158 @@ def match_candidates_by_distance(images, exifs, reference, max_neighbors, max_di
         return set()
     max_neighbors = max_neighbors or 99999999
     max_distance = max_distance or 99999999.
-    k = min(len(images), max_neighbors + 1)
+    k = min(len(images_cand), max_neighbors)
+
+    if(images_ref == images_cand):
+        k = k + 1
```

Comment:
It can happen that `image_ref` intersects with `images_cand` and we still want `k+1` candidates since the first one will be the main image.
Maybe we can always use `k+1` and deal with the exact number of candidates later when we build the list of pairs.